src/services/goToDefinition.ts

```diff
@@ -50,16 +50,7 @@ namespace ts.GoToDefinition {
         // assignment. This case and others are handled by the following code.
         if (node.parent.kind === SyntaxKind.ShorthandPropertyAssignment) {
             const shorthandSymbol = typeChecker.getShorthandAssignmentValueSymbol(symbol.valueDeclaration);
-            if (!shorthandSymbol) {
-                return [];
-            }
-
-            const shorthandDeclarations = shorthandSymbol.getDeclarations();
-            const shorthandSymbolKind = SymbolDisplay.getSymbolKind(typeChecker, shorthandSymbol, node);
-            const shorthandSymbolName = typeChecker.symbolToString(shorthandSymbol);
-            const shorthandContainerName = typeChecker.symbolToString(symbol.parent, node);
-            return map(shorthandDeclarations,
-                declaration => createDefinitionInfo(declaration, shorthandSymbolKind, shorthandSymbolName, shorthandContainerName));
+            return shorthandSymbol ? shorthandSymbol.declarations.map(decl => createDefinitionInfo(decl, typeChecker, shorthandSymbol, node)) : emptyArray;

```

Comment:
if you are using `emptyArray` the return type should be `ReadonlyArray` to avoid messing up the whole system when emptyArray is not empty.. I think `emptyArray` should be types as `ReadOnlyArray<any>` but that would be a separate change and potentially a big one.

Reply:
See #16312 -- will change to `[]` for now.